de.metas.contracts/src/main/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess.java

```diff
@@ -86,102 +77,104 @@ protected String getImportOrderBySql()
-					+ " from " + I_C_BPartner.Table_Name + " bp "
-					+ " where bp." + I_C_BPartner.COLUMNNAME_Value + "=i." + I_I_Flatrate_Term.COLUMNNAME_BPartnerValue
-					+ " and bp." + I_C_BPartner.COLUMNNAME_AD_Client_ID + "=i." + I_I_Flatrate_Term.COLUMNNAME_AD_Client_ID;
-			final String sql = "UPDATE " + I_I_Flatrate_Term.Table_Name + " i "
-					+ "\n SET " + I_I_Flatrate_Term.COLUMNNAME_C_BPartner_ID + "=(" + sqlSelectByValue + ")"
-					+ "\n WHERE " + sqlImportWhereClause
-					+ "\n AND i." + I_I_Flatrate_Term.COLUMNNAME_C_BPartner_ID + " IS NULL";
-
-			final int no = DB.executeUpdateEx(sql, trxName);
-			log.debug("Set C_BPartner_ID for {} records", no);
-
-			// Flag missing BPartners
-			markAsError("BPartner not found", I_I_Flatrate_Term.COLUMNNAME_C_BPartner_ID + " IS NULL"
-					+ "\n AND " + sqlImportWhereClause);
-		}
-
-		//
-		// Update DropShip_BPartner_ID
-		{
-			final String sqlSelectByValue = "select MIN(bp." + I_C_BPartner.COLUMNNAME_C_BPartner_ID + ")"
-					+ " from " + I_C_BPartner.Table_Name + " bp "
-					+ " where bp." + I_C_BPartner.COLUMNNAME_Value + "=i." + I_I_Flatrate_Term.COLUMNNAME_DropShip_BPartner_Value
-					+ " and bp." + I_C_BPartner.COLUMNNAME_AD_Client_ID + "=i." + I_I_Flatrate_Term.COLUMNNAME_AD_Client_ID;
-			final String sql = "UPDATE " + I_I_Flatrate_Term.Table_Name + " i "
-					+ "\n SET " + I_I_Flatrate_Term.COLUMNNAME_DropShip_BPartner_ID + "=(" + sqlSelectByValue + ")"
-					+ "\n WHERE " + sqlImportWhereClause
-					+ "\n AND i." + I_I_Flatrate_Term.COLUMNNAME_DropShip_BPartner_ID + " IS NULL";
-
-			final int no = DB.executeUpdateEx(sql, trxName);
-			log.debug("Set DropShip_BPartner_ID for {} records", no);
-
-			// Flag missing DropShip BPartners
-			markAsError("DropShip BPartner not found",
-					I_I_Flatrate_Term.COLUMNNAME_DropShip_BPartner_ID + " IS NULL"
-							+ "\n AND " + I_I_Flatrate_Term.COLUMNNAME_DropShip_BPartner_Value + " IS NOT NULL" // only where the we have a key to match
-							+ "\n AND " + sqlImportWhereClause);
-		}
-
-		//
-		// Update C_Flatrate_Conditions_ID
-		{
-			final String sqlSelectByValue = "select MIN(fc." + I_C_Flatrate_Conditions.COLUMNNAME_C_Flatrate_Conditions_ID + ")"
-					+ " from " + I_C_Flatrate_Conditions.Table_Name + " fc "
-					+ " where fc." + I_C_Flatrate_Conditions.COLUMNNAME_Name + "=i." + I_I_Flatrate_Term.COLUMNNAME_C_Flatrate_Conditions_Value
-					+ " and fc." + I_C_Flatrate_Conditions.COLUMNNAME_AD_Client_ID + "=i." + I_I_Flatrate_Term.COLUMNNAME_AD_Client_ID;
-			final String sql = "UPDATE " + I_I_Flatrate_Term.Table_Name + " i "
-					+ "\n SET " + I_I_Flatrate_Term.COLUMNNAME_C_Flatrate_Conditions_ID + "=(" + sqlSelectByValue + ")"
-					+ "\n WHERE " + sqlImportWhereClause
-					+ "\n AND i." + I_I_Flatrate_Term.COLUMNNAME_C_Flatrate_Conditions_ID + " IS NULL";
-
-			final int no = DB.executeUpdateEx(sql, trxName);
-			log.debug("Set C_Flatrate_Conditions_ID for {} records", no);
+		dbUpdateBPartnerIds(sqlImportWhereClause);
```

Comment:
Right now this class tells a very deetailed story about how import records are updated and validated and at the same time just a one-liner about how they are imported.
Why not introduce a dedicate class to do the update&validating? I think then it would be more balanced